qiita_db/analysis.py

```diff
@@ -808,81 +808,122 @@ def build_files(self,
+
+                    if rename_dup_samples:
+                        ids_map = {_id: "%d.%s" % (aid, _id)
+                                   for _id in biom_table.ids()}
+                        biom_table.update_ids(ids_map, 'sample', True, True)
+
+                    if new_table is None:
+                        new_table = biom_table
+                    else:
+                        new_table = new_table.merge(biom_table)
 
                 # add the metadata column for study the samples come from,
                 # this is useful in case the user download the bioms
-                study_md = {'Study': artifact.study.title, 'Artifact_id': aid}
-                samples_md = {sid: study_md for sid in selected_samples}
-                biom_table.add_metadata(samples_md, axis='sample')
-                data_type = artifact.data_type
-
-                # this is not checking the reference used for picking
-                # issue #164
-                if new_tables[data_type] is None:
-                    new_tables[data_type] = biom_table
-                else:
-                    new_tables[data_type] = \
-                        new_tables[data_type].merge(biom_table)
+                study_md = {'study': artifact.study.title,
+                            'artifact_ids': ', '.join(artifact_ids),
+                            'reference_id': reference_id,
+                            'command_id': command_id}
+                samples_md = {sid: study_md for sid in new_table.ids()}
+                new_table.add_metadata(samples_md, axis='sample')
 
-            # add the new tables to the analysis
-            _, base_fp = qdb.util.get_mountpoint(self._table)[0]
-            for dt, biom_table in viewitems(new_tables):
-                if biom_table is None:
-                    continue
-                # rarefy, if specified
                 if rarefaction_depth is not None:
-                    biom_table = biom_table.subsample(rarefaction_depth)
+                    new_table = new_table.subsample(rarefaction_depth)
+                    if len(new_table.ids()) == 0:
+                        raise RuntimeError(
+                            "All samples filtered out due to rarefacion level")
+
                 # write out the file
-                biom_fp = join(base_fp, "%d_analysis_%s.biom" % (self._id, dt))
+                fn = "%d_analysis_dt-%s_r-%s_c-%s.biom" % (
+                    self._id, data_type, reference_id, command_id)
+                biom_fp = join(base_fp, fn)
                 with biom_open(biom_fp, 'w') as f:
-                    biom_table.to_hdf5(f, "Analysis %s Datatype %s" %
-                                       (self._id, dt))
-                self._add_file("%d_analysis_%s.biom" % (self._id, dt),
-                               "biom", data_type=dt)
+                    new_table.to_hdf5(
+                        f, "Analysis %d Datatype %s Reference %s Command %s" %
```

Comment:
If I understand correctly, this information is stored as sample metadata. I think it is ok to leave here too if you want, but should we also add "QIITA" at the beginning since that parameter is called "Generated by"?